saleor/graphql/shipping/dataloaders.py

```diff
@@ -18,7 +18,9 @@ class ShippingMethodsByShippingZoneIdLoader(DataLoader):
     context_key = "shippingmethod_by_shippingzone"
 
     def batch_load(self, keys):
-        shipping_methods = ShippingMethod.objects.filter(shipping_zone_id__in=keys)
+        shipping_methods = ShippingMethod.objects.filter(
+            shipping_zone_id__in=keys
+        ).prefetch_related("zip_codes")
```

Comment:
Zip codes should be fetched with a separate dataloader, I added a comment about it in the resolver. This solution would fetch zip codes always, even when the client doesn't ask for this field.